portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -70,4 +82,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private String convertToJsonUsingGson(List<String> comments) {
     return new Gson().toJson(comments);
   }
+
+  private int getNumComments(HttpServletRequest request) { 
+    String numCommentsString = request.getParameter("numComments");
+
+    // Convert the input to an int.
+    int numComments;
+    try {
+      numComments = Integer.parseInt(numCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + numCommentsString);
+      return -1;
```

Comment:
I suspect the walkthrough told you to use -1, but I think it's better practice here to use `Integer.MIN_VALUE` instead (the most negative number that is still an integer). It frees up callers to accept a value of -1.

This is called a "helper method" (or sometimes a "private helper method"). You don't want to put too much logic in here and as much as possible leave the actual "decisions" to the callsites. In this case, right here in this method we don't care if the input value is -1 (it will parse correctly), but if it is and you also return -1 in the error state, a caller can't tell the difference.